### PR TITLE
Angular markdown widget 

### PR DESCRIPTION
**Pull request by japie1235813**
Clean up pr #3934

* Motivation for features / changes
Need to render markdown as html in the text plugin frontend, this widget makes that possible using Angular and marked.js
* Technical description of changes
Added markdown_renderer to widget folder. Provides css to make markdown tables styling consistent with rest of TensorBoard.
* Screenshots of UI changes
https://user-images.githubusercontent.com/28930753/89076662-41aee400-d370-11ea-9f44-99c270a67256.png

* Detailed steps to verify changes work correctly (as executed by you)
Import markdown module into angular component and place `<markdown-renderer [markdown]="YOUR_MARKDOWN_STRING">` to view `YOUR_MARKDOWN_STRING` as HTML in your component.

### CODE REVIEW

File: tensorboard/webapp/widgets/markdown_renderer/markdown_renderer_component.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  selector: 'markdown-renderer',
+  templateUrl: './markdown_renderer_component.ng.html',
+  styleUrls: ['./markdown_renderer_component.css'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class MarkdownRendererComponent {
+  @Input()
+  markdown!: string;
+
+  convertToHTML(markdown: string) {
```
Comment (stephanwlee) [inline tensorboard/webapp/widgets/markdown_renderer/markdown_renderer_component.ts:28]: Could you type the return type? I believe it is string.

**Review by stephanwlee — COMMENTED** [review]
I believe this requires some sync changes. Could you test the sync and make the adjustments accordingly?

**Review by stephanwlee — COMMENTED** [review]
LGTM but can you please test the internal sync before submitting it?

File: tensorboard/webapp/widgets/markdown_renderer/markdown_renderer_component_test.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  function getComponent(fixture: ComponentFixture<TestableComponent>) {
+    return fixture.nativeElement.querySelector('markdown-renderer');
+  }
+
+  it('renders markdown into html', async () => {
+    const fixture = createComponent();
+    fixture.componentInstance.content = '# title';
+    await fixture.detectChanges();
```
Comment (stephanwlee) [inline tensorboard/webapp/widgets/markdown_renderer/markdown_renderer_component_test.ts:53]: `detectChanges` does not return promise so it should not be await.

Comment (japie1235813) [reply tensorboard/webapp/widgets/markdown_renderer/markdown_renderer_component_test.ts]: Thanks! Please help review agin.

**Review by stephanwlee — APPROVED** [review]

**Pull request merged**